### PR TITLE
added the boundsRangeLow and BoundsRangeHigh logic

### PR DESCRIPTION
**Pull request by mukhtar-abdul-ghani**

### CODE REVIEW

File: coordinator/src/services/translation/request/dosage.ts [hunk 1]
```diff
@@ -581,19 +581,29 @@ function getListWithSeparators(list: Array<string>) {
...
+  const element = []
+  if(lowUnit !== highUnit && highQuantity && lowQuantity){
+    element.push(lowValue, " ", lowUnit, " to ", highValue, " ", highUnit)
   }
-  elements.push(" to ", stringifyQuantityValue(highQuantity), " ", highUnit)
-  return elements
+  if(lowUnit === highUnit && highQuantity && lowQuantity){
+    element.push(lowValue, " to ", highValue, " ", highUnit)
+  }
+  if(!highUnit){
```
Comment (dan-jones-iw) [inline coordinator/src/services/translation/request/dosage.ts:598]: can we make this a bit clearer? maybe `if (lowQuantity && !highUnit)`

Comment (dan-jones-iw) [reply coordinator/src/services/translation/request/dosage.ts:598]: same on 601

File: coordinator/src/services/translation/request/dosage.ts [hunk 2]
```diff
@@ -581,19 +581,29 @@ function getListWithSeparators(list: Array<string>) {
...
   const highQuantity = range?.high
   const lowUnit = stringifyQuantityUnit(lowQuantity, pluralise)
   const highUnit = stringifyQuantityUnit(highQuantity, pluralise)
-  const elements = [stringifyQuantityValue(lowQuantity)]
-  if (lowUnit !== highUnit) {
-    elements.push(" ", lowUnit)
+  const lowValue = stringifyQuantityValue(lowQuantity)
+  const highValue = stringifyQuantityValue(highQuantity)
+  const element = []
+  if(lowUnit !== highUnit && highQuantity && lowQuantity){
```
Comment (dan-jones-iw) [inline coordinator/src/services/translation/request/dosage.ts:592]: could you sort out the whitespacing on this line? (`if (` & `) {`)

Comment (dan-jones-iw) [reply coordinator/src/services/translation/request/dosage.ts:592]: same on other lines

File: coordinator/src/services/translation/request/dosage.ts [hunk 3]
```diff
@@ -581,19 +581,29 @@ function getListWithSeparators(list: Array<string>) {
...
+    element.push(lowValue, " ", lowUnit, " to ", highValue, " ", highUnit)
```
[the remark this answers is not in the document]
Comment (mukhtar-abdul-ghani) [reply coordinator/src/services/translation/request/dosage.ts:593]: done

**Review by dan-jones-iw — APPROVED** [review]

File: coordinator/src/services/translation/request/dosage.ts [hunk 4]
```diff
@@ -581,17 +581,24 @@ function getListWithSeparators(list: Array<string>) {
...
+  if (highQuantity && !lowQuantity) {
+    return ["up to ", highValue, " ", highUnit]
+  }
   if (lowUnit !== highUnit) {
-    elements.push(" ", lowUnit)
+    return [lowValue, " ", lowUnit, " to ", highValue, " ", highUnit]
   }
-  elements.push(" to ", stringifyQuantityValue(highQuantity), " ", highUnit)
-  return elements
+  return [lowValue, " to ", highValue, " ", highUnit]
```
Comment (JackRobinsonIW) [inline coordinator/src/services/translation/request/dosage.ts:601]: is there any reason we actually need to return this split array? could we not do something like 

`[${lowValue} to ${highValue} ${highUnit}]`

Comment (JackRobinsonIW) [reply coordinator/src/services/translation/request/dosage.ts:601]: just realised this wouldn't error if `lowValue/lowUnit/highValue/highUnit = null` so I think its better to leave as it

**Pull request merged**